packages/core/src/actors/observable.ts

```diff
@@ -61,7 +72,7 @@ export function fromObservable<T, TEvent extends EventObject>(
           return state;
         case stopSignalType:
           state.canceled = true;
-          state.subscription!.unsubscribe();
+          state.subscription?.unsubscribe();
```

Comment:
when the subscription can be missing when receiving `stopSignalType`? This sounds like an impossible state

Reply:
Ah, `stopSignalType` is coming from an external `.stop()`, hmmm. It might be needed to have some guard in place here for this case - but I wonder about the exact scenario related to this because it seems that we don't have a test for it.

Reply:
If you restore the observable actor and it's already completed, `subscription` will be undefined.

Reply:
Yes, but this line of code is specifically called at some point - and we should have a test case that would hit this line of the code when the subscription is absent (and we dont have such rn)

Reply:
I can but isn't `?.` safer? I'm not sure what should be tested here since the code won't fail regardless of if it's defined or not. We would be testing the change from `!` to `?` which is sorta meaningless.

Do we need to test all the places where we have `?.`?

Reply:
> I can but isn't ?. safer? 

It is. But it's also potentially a little bit of redundant code, runtime overhead etc. I find defensive programming often obscures the meaning of the code and covers up potential bugs.

For example, from my PoV the important question here is - can the `stopSignalType` even be received when `state.subscription === undefined`? If the answer is "no" but it happens anyway then this `?.` will "swallow" the logical/integration bug and we won't even learn about it. If the answer is "yes" then we can write a test case that would exercise this scenario and we can verify there that nothing throws (and then we will be sure that `?.` is actually needed!)

Reply:
I reverted this